pkg/controller/installation/core_controller.go

```diff
@@ -1446,6 +1448,7 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	} else {
 		instance.Status.ImageSet = imageSet.Name
 	}
+	instance.Status.CalicoVersion = calicoVersion
```

Comment:
I think we should have some unit test with an "existing installation + status" that then reconciles and verifies that the version has been updated so simulate an upgrade behaviour.